lib/schemaUtils.js

```diff
@@ -1386,27 +1427,38 @@ module.exports = {
   /**
    * Create parameters specific for a request
    * @param {*} localParams parameters array
+   * @param {object} options - a standard list of options that's globally passed around. Check options.js for more.
    * @returns {Object} with three arrays of query, header and path as keys.
    * @no-unit-test
    */
-  getParametersForPathItem: function(localParams) {
-    var tempParam,
+  getParametersForPathItem: function(localParams, options) {
+    let tempParam,
       params = {
         query: [],
         header: [],
         path: []
-      };
+      },
+      includeDeprecated = options.includeDeprecated !== undefined ?
```

Comment:
Can be simplified with just `options.includeDeprecated !== false`. This also takes care of invalid values by any chance if there.

Reply:
Done